packages/@react-aria/interactions/src/useFocusWithin.ts

```diff
@@ -56,30 +57,74 @@ export function useFocusWithin(props: FocusWithinProps): FocusWithinResult {
       }
 
       state.isFocusWithin = true;
+
+      // Use native events to handle the onBlur. React does not fire onBlur when an element is disabled.
+      // https://github.com/facebook/react/issues/9142
+      let SyntheticEvent: any = e.constructor;
+      let observer: MutationObserver;
+      let onBlur = (e: FocusEvent) => {
+        // We don't want to trigger onBlurWithin and then immediately onFocusWithin again
+        // when moving focus inside the element. Only trigger if the currentTarget doesn't
+        // include the relatedTarget (where focus is moving).
+        if (state.isFocusWithin && !(e.currentTarget as Element).contains(e.relatedTarget as Element)) {
+          state.isFocusWithin = false;
+
+          // For backwards compatibility, construct a React synthetic event.
+          let event: ReactFocusEvent = new SyntheticEvent('onBlur', 'blur', null, e, e.target);
+          event.currentTarget = e.currentTarget as Element;
+
+          onBlurWithin.current?.(event);
+
+          if (event.isPropagationStopped()) {
+            e.stopPropagation();
+          }
+
+          e.currentTarget.removeEventListener('focusout', onBlur);
+        }
+
+        if (observer) {
+          observer.disconnect();
+          observer = null;
+        }
+      };
+
+      e.currentTarget.addEventListener('focusout', onBlur);
+
+      // If this is a <button> or <input>, use a MutationObserver to watch for the disabled attribute.
+      // This is necessary because Firefox does not fire blur/focusout in this case. We dispatch these events ourselves instead.
+      // For browsers that do, focusout fires before the MutationObserver, so onBlur should not fire twice.
+      if (e.target instanceof HTMLButtonElement || e.target instanceof HTMLInputElement) {
+        let target = e.target;
+        observer = new MutationObserver(() => {
+          if (state.isFocusWithin && target.disabled) {
```

Comment:
what happens if the element is unmounted while focus is on it? It looks like the mutation observer will be left around? not sure if it fires when the element it's observing is about to be removed